src/systems/world-update.js

```diff
@@ -0,0 +1,192 @@
+const zeroPos = new THREE.Vector3(0, 0, 0);
+const zeroQuat = new THREE.Quaternion();
+const oneScale = new THREE.Vector3(1, 1, 1);
+const identity = new THREE.Matrix4();
+identity.identity();
+
+AFRAME.registerSystem("world-update", {
+  init() {
+    this._patchRenderFunc();
+    this._patchThreeJS();
+    this.frame = 0;
+  },
+
+  _patchThreeJS: function() {
+    THREE.Object3D.prototype.getWorldPosition = function(target) {
+      if (target === undefined) {
+        console.warn("THREE.Object3D: .getWorldPosition() target is now required");
+        target = new THREE.Vector3();
+      }
+
+      this.updateMatrices();
+
+      return target.setFromMatrixPosition(this.matrixWorld);
+    };
+
+    THREE.Object3D.prototype.getWorldQuaternion = (function() {
+      const position = new THREE.Vector3();
+      const scale = new THREE.Vector3();
+
+      return function getWorldQuaternion(target) {
+        if (target === undefined) {
+          console.warn("THREE.Object3D: .getWorldQuaternion() target is now required");
+          target = new THREE.Quaternion();
+        }
+
+        this.updateMatrices();
+        this.matrixWorld.decompose(position, target, scale);
+
+        return target;
+      };
+    })();
+
+    THREE.Object3D.getWorldScale = (function() {
+      const position = new THREE.Vector3();
+      const quaternion = new THREE.Quaternion();
+
+      return function getWorldScale(target) {
+        if (target === undefined) {
+          console.warn("THREE.Object3D: .getWorldScale() target is now required");
+          target = new THREE.Vector3();
+        }
+
+        this.updateMatrices();
+        this.matrixWorld.decompose(position, quaternion, target);
+
+        return target;
+      };
+    })();
+
+    THREE.Object3D.prototype.updateMatrix = function() {
+      this.matrix.compose(
+        this.position,
+        this.quaternion,
+        this.scale
+      );
+
+      this.matrixWorldNeedsUpdate = true;
+
+      if (!this.matrixIsModified) {
+        this.matrixIsModified = true;
+
+        if (this.cachedMatrixWorld) {
+          this.cachedMatrixWorld.copy(this.matrixWorld);
+          this.matrixWorld = this.cachedMatrixWorld;
+        }
+      }
+    };
+
+    THREE.Object3D.prototype.applyMatrix = function(matrix) {
+      this.matrix.multiplyMatrices(matrix, this.matrix);
```

Comment:
may want to call the original function here instead to make it clear that these 2 lines are not part of the patch. Or at least block it out with comments.

```js
  THREE.Object3D.prototype.applyMatrix = (function(origApplyMatrix) {
    return function(matrix) {
      origApplyMatrix.call(this, matrix);
      this.matrixWorldNeedsUpdate = true;
      if (!this.matrixIsModified) {
        this.matrixIsModified = true;
        if (this.cachedMatrixWorld) {
          this.cachedMatrixWorld.copy(this.matrixWorld);
          this.matrixWorld = this.cachedMatrixWorld;
        }
      }
    }
  })(THREE.Object3D.prototype.applyMatrix)
```